File: backend/app/utils/grade_utils.py
import re
from typing import Union

def normalize_grade(grade: Union[str, int]) -> str:
    """
    Normalize grade format to "Grade X" format.
    
    Args:
        grade: The grade value as a string or integer
        
    Returns:
        Normalized grade in "Grade X" format
        
    Examples:
        >>> normalize_grade("10")
        "Grade 10"
        >>> normalize_grade("9th")
        "Grade 9"
        >>> normalize_grade("8th Grade")
        "Grade 8"
        >>> normalize_grade(7)
        "Grade 7"
    """
    if grade is None:
        return ""
    
    # Convert to string if it's an integer
    grade_str = str(grade)
    
    # Extract the numeric part using regex
    match = re.search(r'(\d+)', grade_str)
    if match:
        grade_number = match.group(1)
        return f"Grade {grade_number}"
    
    # If no numeric part found, return the original string
    return grade_str

def normalize_grade_for_display(grade: Union[str, int]) -> str:
    """
    Normalize grade for display purposes.
    This is an alias for normalize_grade function.
    
    Args:
        grade: The grade value as a string or integer
        
    Returns:
        Normalized grade in "Grade X" format
    """
    return normalize_grade(grade)

File: src/utils/gradeUtils.ts
/**
 * Normalize grade format to "Grade X" format.
 * 
 * @param grade The grade value as a string or number
 * @returns Normalized grade in "Grade X" format
 * 
 * Examples:
 * normalizeGrade("10") -> "Grade 10"
 * normalizeGrade("9th") -> "Grade 9"
 * normalizeGrade("8th Grade") -> "Grade 8"
 * normalizeGrade(7) -> "Grade 7"
 */
export function normalizeGrade(grade: string | number | undefined | null): string {
  if (grade === undefined || grade === null) {
    return "";
  }
  
  // Convert to string if it's a number
  const gradeStr = String(grade);
  
  // Extract the numeric part using regex
  const match = gradeStr.match(/(\d+)/);
  if (match) {
    const gradeNumber = match[1];
    return `Grade ${gradeNumber}`;
  }
  
  // If no numeric part found, return the original string
  return gradeStr;
}

/**
 * Normalize grade for display purposes.
 * This is an alias for normalizeGrade function.
 * 
 * @param grade The grade value as a string or number
 * @returns Normalized grade in "Grade X" format
 */
export function normalizeGradeForDisplay(grade: string | number | undefined | null): string {
  return normalizeGrade(grade);
}